### PR TITLE
Fix add user secrets

### LINKED ISSUE #44150: AddUserSecrets optional parameter not working as expected
### Description

When using the AddUserSecrets method as follows, I would expect an exception to be thrown if the correlated secrets.json file is missing.  However, the internal secrets provider continues to indicate that it is optional, and no exception is thrown.

`config.AddUserSecrets<Startup>(optional: false);`

### Configuration

.NET Core 3.1.9
Windows 10
x64

### Other information

I posted [this issue on StackOverflow](https://stackoverflow.com/questions/64647810/can-the-user-secrets-file-be-declare-as-required-for-an-asp-net-core-applicati) and received confirmation from other users that they are seeing similar behavior.  One user pointed out that [the source itself](https://github.com/dotnet/runtime/blob/1ade5327c94d1b67622705b9f9b76fac4b49e677/src/libraries/Microsoft.Extensions.Configuration.UserSecrets/src/UserSecretsConfigurationExtensions.cs#L192) uses the parameter only for checking the project setting, but not for the existence of an actual secrets file, which contradicts the behavior of the underlying JsonConfigurationProvider.

### PR DESCRIPTION
**Pull request by galakt**
Fixes #44150

### CODE REVIEW

**Comment by galakt** [comment]
@maryamariyan @safern ping

**Comment by maryamariyan** [comment]
@galakt thanks for the PR. will take a look

**Review by maryamariyan — APPROVED** [review]
LGTM. Thanks @galakt

File: src/libraries/Microsoft.Extensions.Configuration.UserSecrets/tests/ConfigurationExtensionTest.cs [hunk 1]
```diff
@@ -107,6 +107,19 @@ public void AddUserSecrets_DoesNotThrowsIfOptional()
...
+        public void AddUserSecrets_DoesThrowsIfNotOptionalAndSecretDoesNotExist()
+        {
+            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
+            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
+            if (File.Exists(secretPath))
+            {
+                File.Delete(secretPath);
+            }
+
+            Assert.Throws<FileNotFoundException>(() => new ConfigurationBuilder().AddUserSecrets(Assembly.GetExecutingAssembly(), false).Build());
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Configuration.UserSecrets/tests/ConfigurationExtensionTest.cs:120]: nit: ideally you'd only want to keep the method that you're asserting behavior within the Assert.Throws block.

File: src/libraries/Microsoft.Extensions.Configuration.UserSecrets/src/UserSecretsConfigurationExtensions.cs [hunk 2]
```diff
@@ -169,6 +169,9 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
         /// <param name="reloadOnChange">Whether the configuration should be reloaded if the file changes.</param>
         /// <returns>The configuration builder.</returns>
         public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder configuration, string userSecretsId, bool reloadOnChange)
+            => AddUserSecretsInternal(configuration, userSecretsId, true, reloadOnChange);
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.Configuration.UserSecrets/src/UserSecretsConfigurationExtensions.cs:172]: nit, usually we name bool parameters. It is not a big deal, but for next time. It helps understand what the true is for when reading the call site.

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**